kumascript/macros/PreviousMenuNext.ejs

```diff
@@ -37,29 +35,34 @@ var s_Menu = mdn.localString({
     "zh-TW": " 概述：",
 });
 
-if ($0) {
-    strPrevious = '<li><a class="button secondary" href="/' + lang + '/docs/' + $0.replace(/ /g, '_') + '"><span class="button-wrap">' + s_PreviousNext[0] + '</span></a></li>';
+function generateLink(pageSlug, linkText) {
+  return `<li><a class="button secondary" href="/${locale}/docs/${pageSlug}"><span class="button-wrap">${linkText}</span></a></li>`;
 }
 
-if ($1) {
-    strNext = '<li><a class="button secondary" href="/' + lang + '/docs/' + $1.replace(/ /g, '_') + '"><span class="button-wrap">' + s_PreviousNext[1] + '</span></a></li>';
+function generateMenuLink(pageSlug) {
+  const url = `/${locale}/docs/${pageSlug}`;
+  let aPage = info.getPageByURL(url);
+  if (!aPage.url && locale !== defaultLocale) {
+    const fallbackUrl = `/${defaultLocale}/docs/${pageSlug}`;
+    aPage = info.getPageByURL(fallbackUrl);
+  }
+  
+  let title = "";
+  if (!aPage.url) {
+    // fallback to generate title from slug
+    title = pageSlug.split('/').at(-1).replace(/_/g, " ");
```

Comment:
When's this necessary? When the macro's wrong, and linking to a page which doesn't exist in en-US? Should we throw an error in that case instead, or just hide the Overview link?

Reply:
We do that regularly to link to en-US if a page isn't available in the locale.

Reply:
When the linked document does not exist at all (localized documents as well as English documents). I think providing 
 broken link flaws for authors is better than throwing an exception (which would make the macro completely unrendered, preventing readers from accessing otherwise correct links).

---

For localized documents, sometimes we (localization team members) may not be able to update the macro call in time, but if only one of the links become invalid (for example, the referenced page is deleted) and makes other correct links inaccessible, the UX may get worse.